astropy/table/column.py

```diff
@@ -536,6 +536,8 @@ def __array_finalize__(self, obj):
         if not hasattr(self, 'indices'):  # may have been copied in __new__
             self.indices = []
         self._copy_attrs(obj)
+        if 'info' in getattr(obj, '__dict__', {}):
```

Comment:
This does not appear to be directly tested. Do you have cases in mind that fail in current main?

Reply:
I'm not sure anymore... But on current master:
```
In [4]: mc = MaskedColumn([1, 2, 3], mask=[True, False, True])

In [5]: mc.info.serialize_method
Out[5]: 
{'fits': 'null_value',
 'ecsv': 'null_value',
 'hdf5': 'data_mask',
 None: 'null_value'}

In [6]: mc.info.serialize_method['fits'] = 'data_mask'

In [7]: mc12 = mc[1:]

In [8]: mc12.info.serialize_method
Out[8]: 
{'fits': 'null_value',
 'ecsv': 'null_value',
 'hdf5': 'data_mask',
 None: 'null_value'}
```

Which suggests I should add a test for this!

Reply:
Actually, the `.view` test does go through this path. But since I couldn't remember that, I added a slice tests too.

Reply:
And the test for the slice brought out that yet another fix was needed... Now done.